MaxText/train.py

```diff
@@ -148,14 +148,15 @@ def record_activation_metrics(output_metrics, intermediate_outputs, config):
       output_metrics['scalar'][f'activ_mean/layer_{layer_num:03d}'] = layer["activation_mean"][0]
       output_metrics['scalar'][f'activ_stdev/layer_{layer_num:03d}'] = layer["activation_stdev"][0]
 
-def train_step(model, config, state, data, dropout_rng):
+def train_step(model, config, state, data, dropout_rng, batch_size):
```

Comment:
hmmm shouldn't be adding random arguments. Can config calculate this? (If not, let's have config derive this.)

Reply:
I changed it so that it will call `max_utils.calculate_raw_input_batch_size` to calculate the `batch_size`.